### PR TITLE
New custom flit backend

### PR DESCRIPTION
**Pull request by arthurzam**
Still needs much testing, and comparison that all files exists with similar contents!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+man html:
+	$(SPHINX_BUILD) -a -b $@ doc build/sphinx/$@
+
+.PHONY: sdist wheel
+sdist wheel:
+	$(PYTHON) -m build --$@
+
+.PHONY: clean
+clean:
+	$(RM) -r build/sphinx doc/api
```
Comment (mgorny) [inline Makefile:15]: Don't you need to define `RM`? Also, you want `-f` or it'll fail when dirs don't exist.

File: doc/conf.py [hunk 2]
```diff
@@ -227,25 +226,32 @@
...
 
 # -- Options for manual page output --------------------------------------------
+generate_man(os.path.abspath('..'), '', 'pkgcore')
+generate_html(os.path.abspath('..'), os.path.abspath('../src'), 'pkgcore')
 
-bin_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))), 'bin')
-scripts = os.listdir(bin_path)
+try:
+    import tomllib
+except ImportError:
```
Comment (mgorny) [inline doc/conf.py:234]: Let's do it via `if sys.version_info`, that's better handled by random tools.

File: pyproject.toml [hunk 3]
```diff
@@ -1,11 +1,87 @@
...
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dynamic = ["version"]
+
+dependencies = [
+	"snakeoil~=0.10.0",
```
Comment (mgorny) [inline pyproject.toml:32]: Inconsistent indentation.

**Review by mgorny — COMMENTED** [review]
Just took a quick look, didn't review fully.

File: Makefile (hunk 1 above)
Comment (mgorny) [reply Makefile:15]: Hmm, I didn't know that. I suppose it's fine.

File: pyproject.toml [hunk 4]
```diff
@@ -1,11 +1,85 @@
...
+[tool.flit.sdist]
+include = [
+	"tox.ini", ".coveragerc", "Makefile", "py_build.py",
+	"NEWS.rst", "doc", "tests", "examples", "bin",
+	"build/sphinx/man/*.1", "build/sphinx/man/*.5",
+]
+exclude = [
+	".github/", ".gitignore",
+	"doc/api/", "doc/generated/",
+]
```
Comment (mgorny) [inline pyproject.toml:74]: I personally prefer these one per line but up to you.

File: pyproject.toml [hunk 5]
```diff
@@ -1,11 +1,85 @@
...
+
+[project.entry-points."pytest11"]
```
Comment (mgorny) [inline pyproject.toml:76]: BTW does `pytest11` need to be quoted here? It looks weird but maybe it does.

**Review by mgorny — APPROVED** [review]
I think it's as good as it can get. `pycodestyle` will probably have a few complaints still.

**Comment by arthurzam** [comment]
@mgorny many thanks. Currently the lessons learned from this process are enough for me to do myself `pkgdev`, but for `pkgcheck` I will do some more thinking and open future discussion.

**Pull request merged**